pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -416,17 +421,42 @@ func (d *namespacedResourcesDeleter) deleteEachItem(
 		return nil
 	}
 	apiResource := metav1.APIResource{Name: gvr.Resource, Namespaced: true}
+
+	waitForAllDeletes := sync.WaitGroup{}
```

Comment:
> Don't we have a util helper for this?

For making a wait group or for spawning go funcs which write results to a local type in a channel?